tests/apm_tracing_e2e/test_otel.py

```diff
@@ -26,57 +27,63 @@ def test_datadog_otel_span(self):
-
         # Assert the child sent by the agent.
-        child = _get_span(spans, "child.span.otel")
+        # childName is no longer the operation name, rather the resource name
+        # after remapping the OTel attributes to Datadog semantics
+        child = _get_span(spans, "", "otel-name.dd-resource")
         assert child.get("parentID") == parent.get("spanID")
         assert child.get("spanID") != "10000"
         assert child.get("duration") == "1000000000"
         assert child.get("meta").get("span.kind") == "internal"
-        assert child.get("type") != "internal"
 
         # Assert the spans received from the backend!
         spans = interfaces.backend.assert_request_spans_exist(self.req, query_filter="")
         assert 2 == len(spans), _assert_msg(2, len(spans))
 
     def setup_distributed_otel_trace(self):
-        self.req = weblog.get("/e2e_otel_span/mixed_contrib", {"shouldIndex": 1, "parentName": "parent.span.otel"},)
+        self.req = weblog.get(
+            "/e2e_otel_span/mixed_contrib", {"shouldIndex": 1, "parentName": "root-otel-name.dd-resource"},
+        )
 
     @irrelevant(condition=context.library != "golang", reason="Golang specific test with OTel Go contrib package")
     def test_distributed_otel_trace(self):
         spans = _get_spans_submitted(self.req)
         assert 3 <= len(spans), _assert_msg(3, len(spans), "Agent did not submit the spans we want!")
 
         # Assert the parent span sent by the agent.
-        parent = _get_span(spans, "parent.span.otel")
-        assert parent["name"] == "parent.span.otel"
+        parent = _get_span(spans, "", "root-otel-name.dd-resource")
+        assert parent["name"] == "otel_unknown"
         assert parent.get("parentID") is None
         assert parent["metrics"]["_dd.top_level"] == 1.0
 
         # Assert the Roundtrip child span sent by the agent, this span is created by an external OTel contrib package
-        roundtrip_span = _get_span(spans, "HTTP_GET")
-        assert roundtrip_span["name"] == "HTTP_GET"
+        roundtrip_span = _get_span(spans, "client.request", "")
+        assert roundtrip_span["name"] == "client.request"
+        assert roundtrip_span["resource"] == "HTTP GET"
         assert roundtrip_span.get("parentID") == parent.get("spanID")
 
         # Assert the Handler function child span sent by the agent.
-        handler_span = _get_span(spans, "testOperation")
-        assert handler_span["name"] == "testOperation"
+        handler_span = _get_span(spans, "server.request", "")
+        assert handler_span["resource"] == "testOperation"
         assert handler_span.get("parentID") == roundtrip_span.get("spanID")
 
         # Assert the spans received from the backend!
         spans = interfaces.backend.assert_request_spans_exist(self.req, query_filter="")
         assert 3 == len(spans), _assert_msg(3, len(spans))
 
 
-def _get_span(spans, span_name):
+def _get_span(spans, span_name, resource_name):
```

Comment:
Perhaps, we can split this method in _get_span_by_name and _get_span_by_resource or something like this, in order to easily read the code. It's a proposal....

Reply:
I also merged the main